source/Halibut.Tests/Util/ClientServiceBuilder.cs

```diff
@@ -50,11 +50,6 @@ public ClientServiceBuilder WithServiceFactory(IServiceFactory serviceFactory)
             return this;
         }
 
-        public ClientServiceBuilder WithService<TContract>(TContract implementation)
```

Comment:
This was the trap it would not actually be possible to load the contract unless the interface type was specified.

Reply:
Cool.

🤔 Doesn't the left over `WithService` method still have the same issue?

E.g. would this `.WithService<IEchoService>(() => new EchoService())` still compile fine as this `.WithService(() => new EchoService())` ?

Reply:
Hmm good point, somehow I never fell into that trap. I wonder if the method can be declared such that the type must be an interface and not a class.

Reply:
That would be amazing! But I don't think that's a constraint that can be applied to generics 😞